### PR TITLE
Provide stress majorization algorithm

### PR DESCRIPTION
**Pull request by jiahao**
Here's another algorithm for graph layouts based on Kamada-Kawai's spring force layout algorithm, but reformulated as a stress majorization algorithm by [Gansner, Koren and North, 2005](http://link.springer.com/chapter/10.1007%2F978-3-540-31843-9_25).

I've also written up documentation in `@doc` strings and updated the tests so that the results of drawing are compared against cached renderings.

Note that this algorithm solves a quadratic minimization problem by simple fixed-point iteration. I think some variant of this algorithm that allows constraints on the generated coordinates (e.g. to ensure that nodes and labels don't overlap) will be super useful and right up your alley. :)

### CODE REVIEW

**Comment by IainNZ** [comment]
Haven't reviewed this, how did the first commit end up as mine?
Also just realized that Travis only runs on nightlies, yikes.

File: REQUIRE [hunk 1]
```diff
@@ -1 +1,4 @@
-julia 0.3-
\ No newline at end of file
+julia 0.3
+Docile
+
```
Comment (IainNZ) [inline REQUIRE]: Some blank lines sneaked in here

**Comment by IainNZ** [comment]
Looks great, I've made your life difficult though in the drawing code because I merged that other PR. If you can rebase on that I'll hit the merge button and tag a new version.

**Comment by jiahao** [comment]
I'm really confused as to why the radii have a factor of ~50x difference in the units on Travis vs my local Macbook... 
https://travis-ci.org/IainNZ/GraphLayout.jl/jobs/56505452#L713-L723

**Comment by IainNZ** [comment]
I just cloned your fork, and I also get those test failures. Is your Compose version the release version?

**Comment by jiahao** [comment]
Oh, I'm on master. I guess that might be the issue. That could mean that the output will break once Compose is upgraded though.

**Comment by IainNZ** [comment]
Seems like quite a drastic change in how Compose is working all the same - are the images visually similar?

**Comment by jiahao** [comment]
No, the graphs drawn with the tagged Compose release look fine but the ones on Compose master have nodes that can't be seen. cc: @dcjones unit bug?

screen shot 2015-03-30 at 10 25 55 pm

screen shot 2015-03-30 at 10 26 00 pm

**Comment by jiahao** [comment]
Also it turns out that differences in the RNG between 0.3 and 0.4 alter the output very subtly. The UUIDs are different, as are the final converged points.

I think the best we can do for now is to run the cache comparisons on 0.4 only. Should be good to merge now

**Comment by dcjones** [comment]
> No, the graphs drawn with the tagged Compose release look fine but the ones on Compose master have nodes that can't be seen. cc: @dcjones unit bug?

It should work now.

**Pull request merged**